google/cloud/pubsub/internal/tracing_message_batch.cc

```diff
@@ -135,13 +135,14 @@ std::function<void(future<void>)> TracingMessageBatch::Flush() {
 
   auto batch_sink_spans = MakeBatchSinkSpans(std::move(message_spans));
 
-  // Set the batch sink parent span as the active span. This will always be the
-  // first span in the vector.
-  auto async_scope = internal::GetTracer(internal::CurrentOptions())
-                         ->WithActiveSpan(batch_sink_spans.front());
-
-  return [next = child_->Flush(),
+  // Give the batch sink parent span its own context. This means the span is
+  // root span with no parent. The parent span always be the first span in the
```

Comment:
"The first span in the vector is the parent of the other spans in the vector." ?

Reply:
Done